src/index.js

```diff
@@ -7,7 +7,9 @@ const process = (component, ...args) =>
 
 export function composed(components, { renderPropName = 'children' } = {}) {
   function Composed(props) {
-    const render = props.children || props.render;
+    const children =
+      typeof props.children === 'function' ? props.children : props.children[0];
```

Comment:
What if `props.children` is not provided? I think it'll land on the `else` part of the ternary expression and will fail trying to index undefined as an array.

I suggest that you check if `this.props` is what you'd get using preact, and in that case use `props.children[0]`, else use `props.children` as is:

```js
const children = (if props.children looks like preact children) ? props.children[0] : props.children;
```